sdk/src/test/kotlin/com/pulumi/kotlin/options/ResourceOptionsTest.kt

```diff
@@ -0,0 +1,545 @@
+package com.pulumi.kotlin.options
+
+import com.pulumi.core.Output
+import com.pulumi.kotlin.KotlinProviderResource
+import com.pulumi.kotlin.KotlinResource
+import com.pulumi.kotlin.assertResourceTransformationResultEquals
+import com.pulumi.kotlin.extractOutputValue
+import com.pulumi.kotlin.mockKotlinResource
+import io.mockk.mockk
+import io.mockk.unmockkAll
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertAll
+import kotlin.test.assertContentEquals
+import kotlin.time.Duration.Companion.milliseconds
+import com.pulumi.resources.ProviderResource as JavaProviderResource
+import com.pulumi.resources.Resource as JavaResource
+import com.pulumi.resources.ResourceArgs as JavaResourceArgs
+import com.pulumi.resources.ResourceOptions as JavaResourceOptions
+import com.pulumi.resources.ResourceTransformation.Args as JavaResourceTransformationArgs
+import com.pulumi.resources.ResourceTransformation.Result as JavaResourceTransformationResult
+
+internal abstract class ResourceOptionsTest<T : ResourceOptions<*>, B : ResourceOptionsBuilder<T>> {
+
+    abstract suspend fun opts(block: suspend B.() -> Unit): T
```

Comment:
(rather minor)

It's probably a nit, but I'd prefer not to use inheritance in tests like that. An alternative would be to make this class concrete, use `ParametrizedTest` (or something like that) and execute each test for two different `opts` implementations.

Reply:
Done, looks like `@MethodSource` accepts anything. `@ArgumentsSource` looks a bit nicer to me ([click](https://stackoverflow.com/questions/46182118/junit5-parameterized-tests-at-class-level)), but it's more typed and getting it to accept this insane lambda didn't really work for me. Tbh, I think I might have preferred inheritance in this case, but this is also fine